release-notes/v0.5.0.md

```diff
@@ -0,0 +1,95 @@
+
+## Code Editing
+
+### Rename refactoring
+
+When a symbol is renamed, the correponding source file is also renamed along with all the references.
+
+![Rename a class](v0.5.0/renaming.gif)
+
+### IntelliCode
+
+With the latest update of Visual Studio IntelliCode, you see recommendations (items starts with a star) on top of the default member list. Those members are recommended by looking at the current type info and the code context around it. Try the AI-assisted coding experience by installing [Visual Studio IntelliCode](https://marketplace.visualstudio.com/items?itemName=VisualStudioExptTeam.vscodeintellicode).
+
+![IntelliCode for Java](v0.5.0/intellicode.gif)
+
+### Attach missing sources
+
+When you navigate to a class in some library without source code, you can attach the missing source zip/jar uring the context menu “Attach Source”.
+
+![Attach missing sources](v0.5.0/attachsource.gif)
+
+## Debugging
+
+### Logpoints
+
+Logpoints allow you to send output to debug console without editing code. They are different from breakpoints because they do not stop the execution flow of your applications. Check out this blog post for more details on [VS Code logpoints](https://code.visualstudio.com/blogs/2018/07/12/introducing-logpoints-and-auto-attach#_introducing-logpoints) feature.
+
+![Logpoints](v0.5.0/logpoints.gif)
+
+### Run or debug using codelens
+
+Don't wanna bother editing `launch.json`? This was made easy by the debugger extension. Now you can click the codelens on top of the `main` function to run or debug your application.
+
+![Run with Codelens](v0.5.0/launch-with-codelens.gif)
+
+## Testing
+
+### TestNG
+
+TestNG support was added to the newest version of [Java Test Runner](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-test). Now you can work with TestNG in VS Code.
+
+![TestNG](v0.5.0/testng.gif)
+
+### Enhanced JUnit 5 support
+
+The support for JUnit 5 is now broader. `@DisplayName` and `@ParameterizedTest` are now supported. Here are some screenshots:
+
+- `@DisplayName`
+
+  ![@DisplayName](v0.5.0/junit5-displayname.png)
+
+- `@ParameterizedTest`
+
+  ![@ParameterizedTest](v0.5.0/junit5-parameterizedtest.png)
+
+### Lazy load
+
+In the original implementation of Java Test Runner, the test case info is fully loaded during startup. This is no longer true in the latest version. Now, the loading only happens when necessary, e.g. when you expand a project to see the test classes in Test viewlet.
+
+## Dependency Management
+
+The long anticipated [Java Dependency Viewer](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-dependency) was released. Now you have a dedicated view of all the packages, classes, and other resources at your fingertip. [Try it now](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-dependency) if you haven't yet. Have fun!
+
+![Java Dependency Viewer](v0.5.0/dependency-viewer.gif)
```

Comment:
Not to pack the *.gif into the release vsix. Use the github CDN link to reference .gif.

The CDN url is like https://raw.githubusercontent.com/Microsoft/vscode-java-pack/master/release-notes/v0.5.0/dependency-viewer.gif

Reply:
Agree not to pack .gif files, simply add them in `.vscodeignore`.  See https://github.com/Microsoft/vscode-maven/blob/master/.vscodeignore

As for the link, I think `vsce` will do the things for you. In vscode-maven's readme.md, it's simply `images/history.gif`, and in .vsix it's already:
```markdown
![Screenshot](https://github.com/Microsoft/vscode-maven/raw/master/images/history.gif)
```

Reply:
Done Thanks!